### PR TITLE
Rancher Template

### PR DESCRIPTION
**Pull request by ground7**
I've added a basic template for the rancher 2.x docker. Rancher is useful for maintaining Kubernetes clusters both locally and in the cloud. It's nice to host the management interface on an Unraid server and then host the apps you'd prefer didn't go down on the cluster.

### CODE REVIEW

File: templates/rancher.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      </Port>
+    </Publish>
+  </Networking>
+  <Data>
+    <Volume>
+      <HostDir>/mnt/user/appdata/rancher/</HostDir>
+      <ContainerDir>/var/lib/rancher</ContainerDir>
+      <Mode>rw</Mode>
+    </Volume>
+  </Data>
```
Comment (Roxedus) [inline templates/rancher.xml:36]: There is unnessecary data here.
https://selfhosters.net/docker/templating/templating/#114-shave-off-the-xml

File: templates/rancher.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+<Container version="2">
+  <Name>rancher</Name>
+  <Repository>rancher/rancher:latest</Repository>
+  <Registry>https://hub.docker.com/r/rancher/rancher</Registry>
+  <Network>bridge</Network>
+  <Privileged>false</Privileged>
+  <Support>https://rancher.com/docs/rancher/v2.x/en/installation/other-installation-methods/single-node-docker/</Support>
+  <Project>https://github.com/rancher/rancher</Project>
+  <Overview>A container management platform built for organizations that deploy containers in production. The docker version of Rancher cannot be made HA (Highly Available), however the clusters it creates can be. It is also possible to connect to and use clusters created by rancher when it is down. Useful as a management interface for deploying, scaling, and editting apps on other systems. (in the cloud, a raspberry pi cluster, vms running k3os, etc.) Perfect for running and managing the apps that you'd rather not go down while maintaining your server.</Overview>
+  <Category/>
```
Comment (Roxedus) [inline templates/rancher.xml:11]: Please add a category
https://selfhosters.net/docker/templating/templating/#21-categories

File: templates/rancher.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <Volume>
+      <HostDir>/mnt/user/appdata/rancher/</HostDir>
+      <ContainerDir>/var/lib/rancher</ContainerDir>
+      <Mode>rw</Mode>
+    </Volume>
+  </Data>
+  <Config Name="Web UI" Target="80" Default="80" Mode="tcp" Description="Container Port: 80" Type="Port" Display="always" Required="true" Mask="false">3080</Config>
+  <Config Name="Secure Web UI" Target="443" Default="443" Mode="tcp" Description="Container Port: 443" Type="Port" Display="always" Required="true" Mask="false">3443</Config>
+  <Config Name="Datastore" Target="/var/lib/rancher" Default="/mnt/user/appdata/rancher/" Mode="rw" Description="Container Path: /var/lib/rancher" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/rancher/</Config>
+</Container>
```
Comment (Roxedus) [inline templates/rancher.xml:40]: Dont add "data" to the config, like the ports 3080 and 3443. Let the user choose the ports.
https://selfhosters.net/docker/templating/templating/#122-add-a-config

File: templates/rancher.xml (hunk 1 above)
Comment (ground7) [reply templates/rancher.xml]: Fixed in 35cb5b7fbf90b3a490eaf2e6508d544221657f09

File: templates/rancher.xml (hunk 3 above)
Comment (ground7) [reply templates/rancher.xml]: Fixed in bcd9058daaf95da19ff9b08aed28c5b66f1c801d

File: templates/rancher.xml (hunk 2 above)
Comment (ground7) [reply templates/rancher.xml]: Fixed in ebc43c1d1b4c9a87b65f84a84f4a299fa7394786

**Pull request merged**